Sources/MapGL/Model/FloorPlan.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+
+/// Represents a floor plan of some building.
+public class FloorPlan {
+	/// An id of a floor plan.
+	public let id: String
+	/// Levels list.
+	public let levels: [String]
+	/// A current level name of a floor plan.
+	public private(set) var currentLevel: String
+
+	/// A current level index of a floor plan.
+	public var currentLevelIndex: Int {
+		get {
+			return self._currentLevelIndex
+		}
+		set {
+			guard newValue >= 0 && newValue < self.levels.count else {
+				let errorText = "Invalid floor level index"
+				assertionFailure(errorText)
+				return
+			}
+			self._currentLevelIndex = newValue
+			self.currentLevel = self.levels[currentLevelIndex]
```

Comment:
Тут странное место. Из сеттера обращаешься к геттеру без нужды.
```suggestion
			self.currentLevel = self.levels[newValue]
```

Reply:
Действительно. Ступил